resources/validations/validate_with_schematron.sh

```diff
@@ -0,0 +1,38 @@
+if [ ! -e  "$1" ]; then
+    echo "no file input for report, exiting"
+    exit 1
+fi
+DOC_TO_VALIDATE="$1"
+echo "doc requested to be validated: ${DOC_TO_VALIDATE}"
+
+# Delete pre-existing XSLT report
+rm -rf target/*.xsl;
+SAXON_VERSION=${SAXON_VERSION:-10.2}
+
+mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.1:get \
+    -DrepoUrl=https://mvnrepository.com/ \
+    -DartifactId=Saxon-HE \
+    -DgroupId=net.sf.saxon \
+    -Dversion=${SAXON_VERSION}
+
+# Delete pre-existing SVRL report
+rm -rf report/schematron/*.results.xml
+
+for qualifiedSchematronName in src/*.sch; do
+    [ -e "${qualifiedSchematronName}" ] || continue
+        
+    # compute name without .sch
+    schematronName=${qualifiedSchematronName##*/}
+    schematronRoot=${schematronName%.*}
+    
+    # Use Saxon XSL transform to convert our Schematron to pure XSL 2.0 stylesheet
+    saxon_jar=~/.m2/repository/net/sf/saxon/Saxon-HE/"${SAXON_VERSION}"/Saxon-HE-"${SAXON_VERSION}".jar
+    java -cp "${saxon_jar}" net.sf.saxon.Transform -o:target/"${schematronRoot}".xsl -s:"${qualifiedSchematronName}" lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl
+    echo "compiling: ${qualifiedSchematronName} to: target/${schematronRoot}.xsl"
+   
+    # Use Saxon XSL transform to use XSL-ified Schematron rules to analyze full FedRAMP-SSP-OSCAL template
+    # and dump the result into reports.
+    reportName="report/schematron/${DOC_TO_VALIDATE}__${schematronRoot}.results.xml"
+    echo "validating doc: ${DOC_TO_VALIDATE} with ${qualifiedSchematronName} output found in "${reportName}
```

Comment:
`shellcheck` recommends:

> [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086): Double quote to prevent globbing and word splitting.

But I think you meant:

```suggestion
    echo "validating doc: ${DOC_TO_VALIDATE} with ${qualifiedSchematronName} output found in ${reportName}"
```

Reply:
> Do we want to keep XSpec separate or conditionally use that too?

I think we want to keep it separate: as an SSP author, I will definately want to run this sh to check an SSP xml I have. As an SSP develeper, I would be interested in running the unit tests. I think there may be eventual needs to add an xspec shell wrapper, but for now, I think those should remain separate.

Reply:
updated